src/generate-grpc-web.ts

```diff
@@ -35,33 +35,46 @@ export function generateGrpcClientImpl(
     assertInstanceOf(methodDesc, FormattedMethodDescriptor);
     chunks.push(code`this.${methodDesc.formattedName} = this.${methodDesc.formattedName}.bind(this);`);
   }
-  chunks.push(code`}\n`);
+  chunks.push(code`}`);
 
   // Create a method for each FooService method
   for (const methodDesc of serviceDesc.method) {
     chunks.push(generateRpcMethod(ctx, serviceDesc, methodDesc));
   }
 
   chunks.push(code`}`);
-  return joinCode(chunks, { trim: false });
+  return joinCode(chunks, { trim: false, on: '\n' });
 }
 
 /** Creates the RPC methods that client code actually calls. */
 function generateRpcMethod(ctx: Context, serviceDesc: ServiceDescriptorProto, methodDesc: MethodDescriptorProto) {
   assertInstanceOf(methodDesc, FormattedMethodDescriptor);
-  const { options, utils } = ctx;
-  const inputType = requestType(ctx, methodDesc);
-  const partialInputType = code`${utils.DeepPartial}<${inputType}>`;
+  const requestMessage = rawRequestType(ctx, methodDesc)
+  const inputType = requestType(ctx, methodDesc, true);
```

Comment:
Instead of invoking both `rawRequestType` + `requestType`, I wonder if `requestType` would return a tuple or something...

Just thinking out loud, b/c we also have an open issue where someone's input type is not actually a message, so we need to make the `fromPartial` call optional, and maybe that'd be nice to handle inside of `requestType` itself... :shrug:

Reply:
Yea I think there is a bit of a code smell that the requestType function manages the wrapping of the type into partial, but it was a minimal change. I suspect it might need a bit of a rethink for that other issue